app/models/reservation.rb

```diff
@@ -0,0 +1,26 @@
+class Reservation < ApplicationRecord
+
+  belongs_to :branch
+  belongs_to :customer
+  has_many :line_items, dependent: :destroy, after_add: :update_total
```

Comment:
what happens if i remove a line item?

Reply:
for that, we have to add a `after_remove` callback too to update the total_count.

Reply:
Or, we need not store total count in table, we can just calculate when sending mails or showing data.

Reply:
after_remove added.